### PR TITLE
Gh6521 2

### PR DESCRIPTION
**Pull request by metas-ts**
#6521

### CODE REVIEW

File: backend/de.metas.business/src/main/java/de/metas/order/impl/OrderLineBL.java [hunk 1]
```diff
@@ -773,6 +773,12 @@ public ProductPrice getCostPrice(final org.compiere.model.I_C_OrderLine orderLin
 				.build();
 	}
 
+	@Override
+	public BigDecimal getPriceActual(final org.compiere.model.I_C_OrderLine orderline)
```
Comment (metas-ts) [inline backend/de.metas.business/src/main/java/de/metas/order/impl/OrderLineBL.java:777]: as said earlier: returning a BigDecimal as a price without any context is dangerous. instead, return `ProductPrice`

File: backend/de.metas.device.scales/src/test/java/de/metas/device/scales/endpoint/TcpConnectionEndPointTest.java [hunk 2]
```diff
@@ -129,8 +129,6 @@ public void run()
 								final String anotherWrongServerReturnString = MockedEndpoint.createWeightString(new BigDecimal(weight + 10)) + ISiscCmd.SICS_CMD_TERMINATOR;
 								out.write(anotherWrongServerReturnString.getBytes(ICmd.SICS_CMD_CHARSET));
 								System.out.println("TcpConnectionEndPointTest" + ": server-socked-thread replied with anotherWrongServerReturnString=" + anotherWrongServerReturnString);
-
-								out.flush();
```
Comment (metas-ts) [inline backend/de.metas.device.scales/src/test/java/de/metas/device/scales/endpoint/TcpConnectionEndPointTest.java:133]: unrelated change that got in here in the context of #6205 .. need to check this myself

File: backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inout/invoicecandidate/M_InOutLine_Handler.java [hunk 3]
```diff
@@ -823,7 +822,7 @@ public static PriceAndTax calculatePriceAndTax(
 				.taxCategoryId(pricingResult.getTaxCategoryId())
 				//
 				.priceEntered(pricingResult.getPriceStd())
-				.priceActual(pricingResult.getPriceStd())
+				.priceActual(inoutLine.getC_OrderLine_ID() != 0 ? Services.get(IOrderLineBL.class).getPriceActual(inoutLine.getC_OrderLine()) : pricingResult.getPriceStd())
```
Comment (metas-ts) [inline backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inout/invoicecandidate/M_InOutLine_Handler.java:825]: we can'T combine a number from a C_OrderLine with an UOM from the prcing engine. there is no guarantee that they fit together

Comment (dragospodariu96) [reply backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inout/invoicecandidate/M_InOutLine_Handler.java]: removed old logic, using the existing getcostprice() method from orderlineBL

File: backend/de.metas.business/src/main/java/de/metas/order/impl/OrderLineBL.java (hunk 1 above)
Comment (dragospodariu96) [reply backend/de.metas.business/src/main/java/de/metas/order/impl/OrderLineBL.java]: removed that completely, using getcostprice() which returns a Productprice with price actual and uomID
